apps/internal/base/base.go

```diff
@@ -345,6 +349,26 @@ func (b Client) AcquireTokenSilent(ctx context.Context, silent AcquireTokenSilen
 	if silent.Claims == "" {
 		ar, err = AuthResultFromStorage(storageTokenResponse)
 		if err == nil {
+			if shouldRefresh(storageTokenResponse.AccessToken.RefreshOn.T) {
```

Comment:
This must be synchronized so that only one goroutine at a time tries to refresh the token, and there should be some backoff--respecting any Retry-After header--between refresh attempts